lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -504,13 +504,22 @@ def run_agent_on(host, arg='--no-daemonize --verbose --onetime --test',
         #                            or a role (String or Symbol) that identifies one or more hosts.
         # @param ip_spec [Hash{String=>String}] a hash containing the host to ip
         #   mappings
-        # @example Stub puppetlabs.com on the master to 127.0.0.1
-        #   stub_hosts_on(master, 'puppetlabs.com' => '127.0.0.1')
-        def stub_hosts_on(machine, ip_spec)
+        # @param alias_spec [Hash{String=>Array[String]] an hash containing the host to alias(es) mappings to apply
+        # @example Stub puppetlabs.com on the master to 127.0.0.1 with an alias example.com
+        #   stub_hosts_on(master, {'puppetlabs.com' => '127.0.0.1'}, {'puppetlabs.com' => ['example.com']})
+        def stub_hosts_on(machine, ip_spec, alias_spec={})
           block_on machine do | host |
             ip_spec.each do |address, ip|
+              aliases = alias_spec[address] || []
+              manifest =<<-EOS.gsub /^\s+/, ""
+                host { '#{address}':
+                  \tensure       => present,
+                  \tip           => '#{ip}',
+                  \thost_aliases => #{aliases},
+                }
+              EOS
               logger.notify("Stubbing address #{address} to IP #{ip} on machine #{host}")
-              on( host, puppet('resource', 'host', address, 'ensure=present', "ip=#{ip}") )
+              apply_manifest_on( host, manifest )
```

Comment:
It looks like this `apply_manifest_on` call is happening once per `ip_spec`, rather than once per host as below.

It looks like this work of creating & applying the manifest is duplicate code. Could you make a helper method that could be called from both of these locations?  Could this happen with the teardown too?